src/codegen/doc/mod.rs

```diff
@@ -524,6 +551,20 @@ where
                 "{}",
                 reformat_doc(&fix_param_names(doc, &self_name), &symbols, &parent_name)
             )?;
+            // fold out parameters as part of returns
+            let out_parameters = fn_.parameters().iter().filter(|param| {
+                !(param.direction == ParameterDirection::Out
```

Comment:
I don't think this one should be inverted? Now you're getting all but `ParameterDirection::Out` :)

Reply:
yeah, already fixed. I only pushed to be able to easily test on gtk4-rs

Reply:
Oh, it should be properly working now. But one very bad thing is that sometimes a functions returns true or false and we convert that to an option which makes the whole thing not very nice I guess. 

See for example this function that returns ` Option<(i32, i32)>`
```rust
    /// [`true`] if `cell` belongs to `self`.
    /// return location for the horizontal
    ///  position of `cell` within `self`, may be [`None`]
    /// return location for the width of `cell`,
    ///  may be [`None`]
```

Reply:
In general out/inout params are just bad for introspection :/

Reply:
Note: if we end up keeping this, we should also handle `n_` cases as they are used as out params sometimes as well.

Reply:
Okay never mind, i actually made it slightly better 
![image](https://user-images.githubusercontent.com/7660997/118892110-b0a9f900-b900-11eb-9e7d-e715ff24f05c.png)

Of course there are some cases that can't be catched and ignored like this one 
![image](https://user-images.githubusercontent.com/7660997/118893055-1e0a5980-b902-11eb-9b1c-84d5659d19bf.png)

but in general, it puts the information where it belongs, in the return instead of parameters.

Reply:
Oh never mind, I just learned about the `length` annotation... It should be fixable.

Reply:
![image](https://user-images.githubusercontent.com/7660997/118898474-e3f28500-b90c-11eb-9e1b-a07983192882.png)
I have pushed the fixed version